sdk/keyvault/keyvault-admin/src/generated/models/index.ts

```diff
@@ -49,37 +61,37 @@ export interface RoleDefinition {
   /**
    * The role type.
    */
-  roleType?: string;
+  roleType?: RoleType;
   /**
    * Role definition permissions.
    */
   permissions?: Permission[];
   /**
    * Role definition assignable scopes.
    */
-  assignableScopes?: string[];
+  assignableScopes?: RoleScope[];
 }
 
 /**
  * Role definition permissions.
  */
 export interface Permission {
   /**
-   * Allowed actions.
+   * Action permissions that are granted.
    */
   actions?: string[];
   /**
-   * Denied actions.
+   * Action permissions that are excluded but not denied. They may be granted by other role definitions assigned to a principal.
    */
   notActions?: string[];
   /**
-   * Allowed Data actions.
+   * Data action permissions that are granted.
```

Comment:
Regarding these comments, I am not sure if I can easily parse what's an `action permission`. Is that how the service docs call this? Feels a bit odd, but it's not like I have a strong bias against it.